projects/power_distribution/src/current_measurement.c

```diff
@@ -1,129 +1,56 @@
 #include "current_measurement.h"
 
-#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
 
-#include "bts7040_load_switch.h"
-#include "bts7200_load_switch.h"
-#include "bts7xxx_common.h"
 #include "log.h"
+#include "output.h"
 
-static PowerDistributionCurrentHardwareConfig s_hw_config;
-static PowerDistributionCurrentStorage s_storage = { 0 };
-static Bts7200Storage s_bts7200_storages[MAX_POWER_DISTRIBUTION_BTS7200_CHANNELS];
-static Bts7040Storage s_bts7040_storages[MAX_POWER_DISTRIBUTION_BTS7040_CHANNELS];
+static CurrentMeasurementConfig *s_config;
+static CurrentMeasurementStorage s_storage = { 0 };
 static SoftTimerId s_timer_id;
 
-static Bts7xxxEnablePin *s_enable_pins[2 * MAX_POWER_DISTRIBUTION_BTS7200_CHANNELS +
-                                       MAX_POWER_DISTRIBUTION_BTS7040_CHANNELS];
-static size_t s_num_enable_pins;
-
 static uint32_t s_interval_us;
-static PowerDistributionCurrentMeasurementCallback s_callback;
+static CurrentMeasurementCallback s_callback;
 static void *s_callback_context;
 
 static void prv_measure_currents(SoftTimerId timer_id, void *context) {
-  // read from all the BTS7200s
-  for (uint8_t i = 0; i < s_hw_config.num_bts7200_channels; i++) {
-    mux_set(&s_hw_config.mux_address, s_hw_config.bts7200s[i].mux_selection);
-    bts7200_get_measurement(&s_bts7200_storages[i],
-                            &s_storage.measurements[s_hw_config.bts7200s[i].current_0],
-                            &s_storage.measurements[s_hw_config.bts7200s[i].current_1]);
-  }
-
-  // read from all the BTS7040s
-  for (uint8_t i = 0; i < s_hw_config.num_bts7040_channels; i++) {
-    mux_set(&s_hw_config.mux_address, s_hw_config.bts7040s[i].mux_selection);
-    bts7040_get_measurement(&s_bts7040_storages[i],
-                            &s_storage.measurements[s_hw_config.bts7040s[i].current]);
+  // read from each output
+  for (uint8_t i = 0; i < s_config->num_outputs_to_read; i++) {
+    Output output = s_config->outputs_to_read[i];
+    if (output >= NUM_OUTPUTS) {
+      // shouldn't be possible, we caught them in init
+      continue;
+    }
+    StatusCode code = output_read_current(output, &s_storage.measurements[output]);
+    if (!status_ok(code)) {
+      LOG_WARN("Could not read current from output %d (index %d): code %d\n", output, i, code);
```

Comment:
GPIO outputs shouldn't be specified in the current measurement config - the abstraction is a bit leaky that way but I think it's ok to spam logs since we won't get valid data from those outputs and we don't want to tx bad data